File: example/java/pact_validation_with_oas/one_consumer_one_provider/consumer/src/test/java/br/com/zup/pact/consumer/stub/ClientStubTest.java
package br.com.zup.pact.consumer.stub;

import br.com.zup.pact.consumer.dto.ClientDetailsDTO;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

@SpringBootTest
public class ClientStubTest {

    @Autowired
    private ClientStub clientStub;

    private static final int NUMBER_OF_STUBS = 10;

    @Test
    void getClientsDetailsDTOStubsListTest() {
        List<ClientDetailsDTO> clientDetailsDTOList = clientStub.getAllStubsDTOFormat();

        clientDetailsDTOList.forEach(clientDetailsDTO -> {
            assertNotNull(clientDetailsDTO.getName());
            assertNotNull(clientDetailsDTO.getFinalName());
            assertNotNull(clientDetailsDTO.getAge());
            assertNotNull(clientDetailsDTO.getAccountId());
            assertNotNull(clientDetailsDTO.getId());
        });

        assertEquals(NUMBER_OF_STUBS, clientDetailsDTOList.size());
    }
}

File: example/java/kotlin/consumer/src/test/kotlin/br/com/zup/pact/consumer/stub/ClientStubTest.kt
package br.com.zup.pact.consumer.stub

import br.com.zup.pact.consumer.dto.ClientDetailsDTO
import br.com.zup.pact.consumer.entity.ClientEntity
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class ClientStubTest() {

    @Autowired
    private lateinit var clientStub: ClientStub

    @Test
    fun `New AccountStub class should has 10 ClientEntity`() {
        val numberOfStubs = 10
        val minAge = 18
        val maxAge = 70

        val actualResult: Map<Int, ClientEntity> = clientStub.clients

        Assertions.assertThat(actualResult).hasSize(numberOfStubs)

        (1..numberOfStubs).forEach { id ->

            val actualEntity = actualResult[id]

            Assertions.assertThat(ClientEntity(id, id, "", "", 18))
                    .usingRecursiveComparison()
                    .ignoringFields("name", "finalName", "age")
                    .isEqualTo(actualEntity)
            Assertions.assertThat(actualEntity?.name).isNotBlank()
            Assertions.assertThat(actualEntity?.finalName).isNotBlank()
            Assertions.assertThat(actualEntity?.age).isBetween(minAge, maxAge)
        }
    }

    @Test
    fun `Method getAllStubsDTOFormat should return the ClientEntity list`() {
        val numberOfStubs = 10
        val minAge = 18
        val maxAge = 70


        val actualResult: List<ClientDetailsDTO> = clientStub.getAllStubsDTOFormat()

        Assertions.assertThat(actualResult).hasSize(numberOfStubs)

        (0 until numberOfStubs).forEach { index ->

            val actualEntity = actualResult[index]

            Assertions.assertThat(ClientDetailsDTO(index + 1, index + 1, "", "", 18))
                    .usingRecursiveComparison()
                    .ignoringFields("name", "finalName", "age")
                    .isEqualTo(actualEntity)
            Assertions.assertThat(actualEntity.name).isNotBlank()
            Assertions.assertThat(actualEntity.finalName).isNotBlank()
            Assertions.assertThat(actualEntity.age).isBetween(minAge, maxAge)
        }
    }
}